### PR TITLE
[ARM] `az data-boundary` Add DataBoundary Cmdlets

### PR DESCRIPTION
**Pull request by skadhiresan**
**Related command**

**Description**

This PR enables databoundary commands to create and get databoundary.

**Testing Guide**

[ARM] `az data-boundary show`: Will get the data boundary at a subscription or rg level
[ARM] `az data-boundary show-tenant`: Will get the data boundary at a tenant level
[ARM] `az data-boundary create`: Will put a data boundary at a tenant level (test will create a databoundary on existing databoundary which has already been created and will check for the proper error message)

**History Notes**

[ARM] `az data-boundary show`: Add command to support getting the data boundary at a specified scope
[ARM] `az data-boundary show-tenant`: Add command to support getting the data boundary at a tenant level
[ARM] `az data-boundary create`: Add command to support creating tenant to data boundary

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by yanzhudd** [comment]
> History Notes
> 
> [ARM] az resources data-boundary show
> [ARM] az resources data-boundary show-tenant
> [ARM] az resources data-boundary create

Please note that these history notes will show in the release notes of Azure CLI, please specify the correct command names and corresponding functions, e.g.,
[ARM] `az data-boundary create`: Add command to support creating tenant to data boundary
...

File: src/azure-cli/azure/cli/command_modules/resource/aaz/latest/data_boundary/_create.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+
+@register_command(
+    "data-boundary create",
+)
+class Create(AAZCommand):
+    """Create tenant to data boundary.
+
+    :example: Opt-in to data boundary
+        az data-boundary create --data-boundary EU --default default
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/resource/aaz/latest/data_boundary/_create.py:21]: It is better not to use parameter names that are the same as command name

**Comment by skadhiresan** [comment]
@yanzhudd updated!

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**